src/services/loggerInterface.ts

```diff
@@ -0,0 +1,48 @@
+export interface ILogger {
+    emerg(message: string): void;
+    alert(message: string): void;
+    crit(message: string): void;
+    error(message: string): void;
+    warning(message: string): void;
+    notice(message: string): void;
+    info(message: string): void;
+    debug(message: string): void;
+}
+
+export abstract class Logger implements ILogger {
+    abstract serviceName: string;
+    abstract instanceName : string;
```

Comment:
These 2 are not really needed: it is up to the concrete implementation to decide how to get/use those values.
If we want to keep them then we should make them protected and not public (public class properties are evil ^^)